### PR TITLE
Stripe-recreate-subscriptions

### PR DESCRIPTION
**Pull request by rhimmelbauer**

### CODE REVIEW

File: src/vendor/processors/stripe.py [hunk 1]
```diff
@@ -1402,49 +1409,34 @@ def get_subscription(self, stripe_subscription):
...
             raise Exception(f"Could not create stripe_setup_intent transaction_info: {self.transaction_info}")
-
-        subscription_obj = self.build_subscription(self.invoice.order_items.first(), stripe_payment_method.id)
-        if previous_start_date:
-            trial_days = offer.get_offer_end_date(previous_start_date) - timezone.now()
-            subscription_obj['trial_period_days'] = trial_days.days
-            subscription_obj['billing_cycle_anchor'] = offer.get_offer_end_date(previous_start_date)
+        
+        subscription_obj = self.build_subscription(self.invoice.order_items.first(), payment_method_id)
+        subscription_obj |= subscription_extras
```
Comment (mgcamisa) [inline src/vendor/processors/stripe.py:1439]: not sure what's going on here- is this supposed to add the subscription extras to the object? 
from what i can tell this line equals `subscription_obj = subscription_obj or subscription_extras`

File: src/vendor/templates/vendor/manage/subscription_detail.html [hunk 2]
```diff
@@ -61,19 +61,22 @@ <h4>{{ object.name}}</h4>
...
-            <form id="cancelSubscription" class="align-itmes-center" method="post"
+            <form id="cancelSubscription" class="align-itmes-center m-2" method="post"
               action="{% url 'vendor_api:customer-subscription-cancel' object.uuid %}">
               {% csrf_token %}
               <button class="btn btn-sm btn-primary" type="submit">{% trans 'Cancel Subscription' %}</button>
             </form>
             {% endif %}
-            <div class="align-itmes-center">
-              <a class="btn btn-primary" href="{% url 'vendor_admin:manager-subscription-add-payment' object.uuid object.profile.uuid %}">{% trans 'Add Payment' %}</a>
+            <div class="align-itmes-center m-2">
```
Comment (mgcamisa) [inline src/vendor/templates/vendor/manage/subscription_detail.html:75]: *items

File: src/vendor/templates/vendor/manage/subscription_detail.html [hunk 3]
```diff
@@ -61,19 +61,22 @@ <h4>{{ object.name}}</h4>
...
+              <a class="btn btn-sm btn-primary" href="{% url 'vendor_admin:manager-subscription-add-payment' object.uuid object.profile.uuid %}">{% trans 'Add Payment' %}</a>
+            </div>
+            <div class="align-itmes-center m-2">
```
Comment (mgcamisa) [inline src/vendor/templates/vendor/manage/subscription_detail.html:78]: *items

File: src/vendor/processors/stripe.py (hunk 1 above)
Comment (rhimmelbauer) [reply src/vendor/processors/stripe.py:1439]: It's a handy way merge two dictionarys

File: src/vendor/templates/vendor/manage/subscription_detail.html (hunk 2 above)
Comment (rhimmelbauer) [reply src/vendor/templates/vendor/manage/subscription_detail.html:75]: thanks :)

**Review by mgcamisa — APPROVED** [review]

**Pull request merged**